### DISCUSSION TITLE
Is the source code for the vscode extension available?

### DISCUSSION THREAD

**Discussion by jimkring** [microsoft/promptflow#360] [Q&A] [answered]
I couldn't find it in the [promptflow](https://github.com/microsoft/promptflow/) repo or elsewhere (after some searching).

Thanks

**Comment by JYC-99** [comment, marked as answer]
There is no plan to open source the ux parts so far. Both web experience and vscode extension code are still in private repos.
We have the library to draw the DAG graph on the GUI public: https://github.com/microsoft/react-dag-editor
For other questions, bug reports or feature suggestions, please open issues and the crew will track them.
